syntax = "proto3";

package api.dwdata.v2;

//import "google/api/annotations.proto";
import "google/protobuf/struct.proto";
option go_package = "micros-dw/api/dwdata/v2;dwdataV2";
option java_multiple_files = true;
option java_outer_classname = "DwdataV2";
option java_package = "api.dwdata.v2";


service DwdataService {

  rpc GetEnterpriseIdent(GetEntIdentReq) returns (GetEntIdentResp) {
  }

  rpc GetEnterpriseInfo(GetEntInfoReq) returns (GetEntInfoResp) {
  }

  rpc GetEnterpriseCredential(GetEntInfoReq) returns (GetEntCredentialResp) {
  }

  rpc GetEnterpriseRankingList(GetEntInfoReq) returns (GetEntRankingListResp) {
  }

  rpc GetEnterpriseIndustry(GetEntInfoReq) returns (GetEntStrArrayResp) {
  }

  rpc GetEnterpriseProduct(GetEntInfoReq) returns (GetEntStrArrayResp) {
  }

  rpc GetEntEquityTransparency(GetEntInfoReq) returns (GetEquityTransparencyResp) {
  }

  rpc GetEntShareholders(GetEntInfoReq) returns (GetShareholdersResp) {
  }

  rpc GetEntInvestment(GetEntInfoReq) returns (GetInvestmentResp) {
  }

  rpc GetEntBranches(GetEntInfoReq) returns (GetBranchesResp) {
  }

}


message Shareholders {
  string shareholder_name = 1;
  string shareholder_type = 2;
  string capital_amount = 3;
  string real_amount = 4;
  string capital_type = 5;
  string percent = 6;
}
message GetShareholdersResp {
  bool success = 1;
  int32 code = 2;
  string msg = 3;
  repeated Shareholders data = 4;
}

message Investment {
  string enterprise_name = 1;
  string operator = 2;
  string shareholding_ratio = 3;
  string invested_amount = 4;
  string start_data = 5;
  string status = 6;
}
message GetInvestmentResp {
  bool success = 1;
  int32 code = 2;
  string msg = 3;
  repeated Investment data = 4;
}


message Branches {
  string enterprise_name = 1;
  string operator = 2;
  string area = 3;
  string start_date = 4;
  string status = 5;
}

message GetBranchesResp {
  bool success = 1;
  int32 code = 2;
  string msg = 3;
  repeated Branches data = 4;
}


message GetEquityTransparencyResp {
  bool success = 1;
  int32 code = 2;
  string msg = 3;
  string conclusion = 4;
  repeated google.protobuf.Struct data = 5;
  string usc_id = 6;
}

message GetEntIdentReq {
  string enterprise_name = 1;
}


message EntIdent {
  bool exists = 3;
  bool isLegal = 4;
  string usc_id = 5;
}

message GetEntIdentResp {
  bool success = 1;
  int32 code = 2;
  string msg = 3;
  EntIdent data = 4;
}

message GetEntInfoResp {
  bool success = 1;
  int32 code = 2;
  string msg = 3;
  EntInfo data = 5;
}

message EntInfo {
  string usc_id = 1;
  string enterprise_title = 2;
  string enterprise_title_en = 3;
  string business_registration_number = 4;
  string establish_date = 5;
  string region = 6;
  string approved_date = 7;
  string registered_address = 8;
  string registered_capital = 9;
  string paid_in_capital = 10;
  string enterprise_type = 11;
  string stuff_size = 12;
  int32 stuff_insured_number = 13;
  string business_scope = 14;
  string import_export_qualification_code = 15;
  string legal_representative = 16;
  string registration_authority = 17;
  string registration_status = 18;
  string taxpayer_qualification = 19;
  string organization_code = 20;
  string url_qcc = 21;
  string url_homepage = 22;
  string business_term_start = 23;
  string business_term_end = 24;
  int64 id = 25;
  string created_at = 26;
  string updated_at = 27;
}

message EntCredential {
  int64 id = 1;
  string usc_id = 2;
  string certification_title = 3;
  string certification_code = 4;
  string certification_level = 5;
  string certification_type = 6;
  string certification_source = 7;
  string certification_date = 8;
  string certification_term_start = 9;
  string certification_term_end = 10;
  string certification_authority = 11;
  string created_at = 12;
  string updated_at = 13;
}

message GetEntCredentialResp {
  bool success = 1;
  int32 code = 2;
  string msg = 3;
  repeated EntCredential data = 4;
}

message GetEntInfoReq {
  string usc_id = 1;
}

message GetEntStrArrayResp {
  bool success = 1;
  int32 code = 2;
  string msg = 3;
  repeated string data = 4;
}

message GetEntRankingListResp {
  bool success = 1;
  int32 code = 2;
  string msg = 3;
  repeated EnterpriseRankingList data = 4;
}

message EnterpriseRankingList {
  string usc_id = 1;
  int32 ranking_position = 2;
  string list_title = 3;
  string list_type = 4;
  string list_source = 5;
  int32 list_participants_total = 6;
  string list_published_date = 7;
  string list_url_qcc = 8;
  string list_url_origin = 9;
}


//    protoc --go_out=. --go_opt=paths=source_relative \
//    --go-grpc_out=. --go-grpc_opt=paths=source_relative \
//    dw.proto
